cli-typescript/src/utils/deployContract.ts

```diff
@@ -438,6 +399,75 @@ export const processStages = async (params: {
   return JSON.parse(stagesData);
 };
 
+const sendERC721SetupTransaction = async ({
+  cm,
+  contractAddress,
+  tokenUriSuffix,
+  uri,
+  maxMintableSupply,
+  globalWalletLimit,
+  mintCurrency,
+  fundReceiver,
+  stagesData,
+  royaltyReceiver,
+  royaltyFee,
+}: {
+  cm: ContractManager;
+  contractAddress: Hex;
+  tokenUriSuffix: string;
+  uri: string;
+  maxMintableSupply: number;
+  globalWalletLimit: number;
+  mintCurrency: string;
+  fundReceiver: string;
+  stagesData: ERC721StageData[];
+  royaltyReceiver: string;
+  royaltyFee: number;
+}) => {
+  try {
+    const setupSignature =
+      'function setup(string,string,uint256,uint256,address,address,(uint80,uint80,uint32,bytes32,uint24,uint256,uint256)[],address,uint96)';
+
+    const abi = AbiFunction.from(setupSignature);
+
+    const parsedStagesData = stagesData.map((stage) => {
+      return [
+        BigInt(stage.price),
+        BigInt(stage.mintFee),
+        stage.walletLimit,
+        stage.merkleRoot as Hex,
+        stage.maxStageSupply,
+        BigInt(stage.startTime),
+        BigInt(stage.endTime),
+      ] as readonly [bigint, bigint, number, Hex, number, bigint, bigint];
```

Comment:
Wonder why we have to double cast types here?

Reply:
The AbiFunction requires precise location-based typings, but the map cmd only returns an array of union types, hence the typecasting.